crates/sui-json-rpc-types/src/sui_transaction.rs

```diff
@@ -1297,6 +1298,70 @@ impl TransactionBytes {
     }
 }
 
+#[serde_as]
+#[derive(Serialize, Deserialize, JsonSchema, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct GaslessTransactionBytes {
+    /// BCS serialized transaction data bytes without its type tag, as base-64 encoded string.
+    pub tx_bytes: Base64,
+}
+
+impl GaslessTransactionBytes {
+    pub fn from_data(data: GaslessTransactionData) -> Result<Self, anyhow::Error> {
+        Ok(Self {
+            tx_bytes: Base64::from_bytes(bcs::to_bytes(&data)?.as_slice()),
+        })
+    }
+
+    pub fn to_data(self) -> Result<GaslessTransactionData, anyhow::Error> {
+        bcs::from_bytes::<GaslessTransactionData>(
+            &self.tx_bytes.to_vec().map_err(|e| anyhow::anyhow!(e))?,
+        )
+        .map_err(|e| anyhow::anyhow!(e))
+    }
+}
+
+#[derive(Clone, Serialize, Deserialize, JsonSchema, Debug)]
+pub struct SponsoredTransactionResponseData {
+    /// BCS serialized transaction data bytes without its type tag, as base-64 encoded string.
+    pub tx_bytes: Base64,
+    /// Base58 encoded transaction digest
+    pub tx_digest: TransactionDigest,
+    /// Base64 encoded transaction signature, signed by the gas owner
+    pub signature: GenericSignature,
+    /// Expiration time of the assigned gas object
+    pub expire_at: u64,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, JsonSchema)]
+pub struct ErrorData {
+    pub details: String,
+}
+#[derive(Debug, Clone, Serialize, Deserialize, JsonSchema)]
+pub struct SponsoredTransactionResponseError {
+    pub code: i32,
+    pub message: String,
+    pub data: ErrorData,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, JsonSchema)]
```

Comment:
Unsure if this is the recommended way to deserialize a response, but I followed the Faucet example in the codebase.